web-wallet/src/lib/components/Allocate/Allocate.svelte

```diff
@@ -54,17 +53,49 @@
 
   let { gasLimit, gasPrice } = gasSettings;
 
-  /** @type {number} */
-  let shieldedAmount = luxToDusk(shieldedBalance);
-
-  /** @type {number} */
-  let unshieldedAmount = luxToDusk(unshieldedBalance);
-
   /** @type {number} */
   let screenWidth = window.innerWidth;
 
   const minAmount = 0.000000001;
 
+  // Internal state of the balances
+  let shielded = shieldedBalance;
+  $: unshielded = totalBalance - shielded;
```

Comment:
Please keep reactive statement as last: it's a bit weird to read the code top to bottom otherwise.
In this case this reactive statement accesses a `totalBalance` variable that isn't defined yet (logically).